modules/core/02-client/keeper/keeper_test.go

```diff
@@ -461,3 +463,73 @@ func (suite KeeperTestSuite) TestIterateClientStates() { //nolint:govet // this
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestUpdateParams() {
+	govAddress := authtypes.NewModuleAddress(govtypes.ModuleName).String()
+
+	testCases := []struct {
+		name     string
+		expPass  bool
+		malleate func(params *types.MsgUpdateParams)
+	}{
+		{
+			name:    "disabling localhost",
+			expPass: true,
+			malleate: func(msg *types.MsgUpdateParams) {
+				msg.Authority = govAddress
+				msg.Params = types.Params{AllowedClients: []string{exported.Tendermint, exported.Solomachine}}
+			},
+		},
+		{
+			name:    "non gov address fails to update params",
+			expPass: false,
+			malleate: func(msg *types.MsgUpdateParams) {
+				msg.Authority = "not-gov-address"
+				msg.Params = types.Params{AllowedClients: types.DefaultAllowedClients}
+			},
+		},
+		{
+			name:    "empty authority fails",
+			expPass: false,
+			malleate: func(msg *types.MsgUpdateParams) {
+				msg.Authority = ""
+				msg.Params = types.Params{AllowedClients: types.DefaultAllowedClients}
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			ctxA := suite.chainA.GetContext()
+			ibcKeeper := suite.chainA.App.GetIBCKeeper()
+			clientParams := ibcKeeper.ClientKeeper.GetParams(ctxA)
+
+			suite.Require().Equal(types.DefaultAllowedClients, clientParams.AllowedClients)
+			suite.Require().True(clientParams.IsAllowedClient(exported.Localhost))
+
+			cs := suite.chainA.GetClientState(exported.Localhost)
+			localhostClientState := cs.(*localhost.ClientState)
+			suite.Require().Equal(clientParams.IsAllowedClient(exported.Localhost), localhostClientState.Enabled)
+
+			msgUpdateParams := &types.MsgUpdateParams{}
+			tc.malleate(msgUpdateParams)
+
+			_, err := ibcKeeper.ClientKeeper.UpdateParams(ctxA, msgUpdateParams)
+
+			clientParams = ibcKeeper.ClientKeeper.GetParams(ctxA)
+			cs = suite.chainA.GetClientState(exported.Localhost)
+			localhostClientState = cs.(*localhost.ClientState)
+
+			suite.Require().Equal(clientParams.IsAllowedClient(exported.Localhost), localhostClientState.Enabled)
```

Comment:
Wouldn't it be more safe to check both `clientParams.IsAllowedClient(exported.Localhost)` and `localhostClientState.Enabled` against an expected value? Otherwise we could have the situation were `UpdateParams` doesn't do what it's supposed to do when disabling the params (neither params get updated nor client state gets updated) and the test would still pass.

Reply:
yes I think that's a good idea!